src/target/ext/register_info.rs

```diff
@@ -0,0 +1,220 @@
+}
+
+impl Format {
+    /// Match an enum field to its corresponding string slice
+    pub fn as_str(&self) -> &str {
+        match self {
+            Format::Binary => "binary",
+            Format::Decimal => "decimal",
+            Format::Hex => "hex",
+            Format::Float => "float",
+            Format::VectorOfSInt8 => "vector-sint8",
+            Format::VectorOfUInt8 => "vector-uint8",
+            Format::VectorOfSInt16 => "vector-sint16",
+            Format::VectorOfUInt16 => "vector-uint16",
+            Format::VectorOfSInt32 => "vector-sint32",
+            Format::VectorOfUInt32 => "vector-uint32",
+            Format::VectorOfFloat32 => "vector-float32",
+            Format::VectorOfUInt128 => "vector-uint128",
+        }
+    }
+}
+
+/// Describes the generic types that most CPUs have
+/// Note:
+#[non_exhaustive]
+pub enum Generic {
+    /// Program counter register
+    Pc,
+    /// Stack pointer register
+    Sp,
+    /// Frame pointer register
+    Fp,
+    /// Return address register
+    Ra,
+    /// CPU flags register Note: The linked documentation above has a typo, correct name is "flags", see [source](https://github.com/llvm/llvm-project/blob/369ce54bb302f209239b8ebc77ad824add9df089/lldb/source/Plugins/Process/gdb-remote/GDBRemoteCommunicationServerLLGS.cpp#L551)
+    Flags,
+    /// Function argument 1
+    Arg1,
+    /// Function argument 2
+    Arg2,
+    /// Function argument 3
+    Arg3,
+    /// Function argument 4
+    Arg4,
+    /// Function argument 5
+    Arg5,
+    /// Function argument 6
+    Arg6,
+    /// Function argument 7
+    Arg7,
+    /// Function argument 8
+    Arg8,
+}
+
+impl Generic {
+    /// Match an enum field to its corresponding string slice
+    pub fn as_str(&self) -> &str {
+        match self {
+            Generic::Pc => "pc",
+            Generic::Sp => "sp",
+            Generic::Fp => "fp",
+            Generic::Ra => "ra",
+            Generic::Flags => "flags",
+            Generic::Arg1 => "arg1",
+            Generic::Arg2 => "arg2",
+            Generic::Arg3 => "arg3",
+            Generic::Arg4 => "arg4",
+            Generic::Arg5 => "arg5",
+            Generic::Arg6 => "arg6",
+            Generic::Arg7 => "arg7",
+            Generic::Arg8 => "arg8",
+        }
+    }
+}
+
+/// This Zero-Sized Type (ZST) token has to be returned from the callback in the
+/// Callback struct below. This is imitating the FnOnce trait.
+pub struct CallbackToken<'a>(pub core::marker::PhantomData<&'a *mut ()>);
+
+/// Wrapper for a callback. The <token> field is our "proof-of-invocation", to
+/// guarantee that the callback stored in the <cb> member has been called.
+pub struct Callback<'a> {
+    /// The callback function that has to be called exactly once
+    pub cb: &'a mut dyn FnMut(Option<Register<'_>>),
+    /// The "proof token" for the callback invocation
+    pub token: CallbackToken<'a>,
+}
+
+/// Trait which acts as a forwarder to the callback in the <cb> field. It will
```

Comment:
Let's workshop this (and other) doc comments to avoid referencing underlying implementation details, and focusing on how/why they are used by the end user.